src/Packetery/Module/Framework/WcAdapter.php

```diff
@@ -52,14 +52,14 @@ public function productFactoryGetProduct( $product_id ) {
 	}
 
 	/**
-	 * Gets product by post id.
+	 * Gets product by post or post id.
 	 *
-	 * @param int $postId Post id.
+	 * @param mixed $theProduct Post id or object.
 	 *
 	 * @return false|\WC_Product|null
 	 */
-	public function getProduct( int $postId ) {
-		return wc_get_product( $postId );
+	public function getProduct( $theProduct ) {
+		return wc_get_product( $theProduct );
```

Comment:
projde tu vůbec string? Proč theProduct? proč ne jen product?

Reply:
volá to vnitřně totiž get_product a podle docu

mixed $product_id WC_Product|WP_Post|int|bool $product Product instance, post instance, numeric or false to use global $post.

Reply:
> projde tu vůbec string? Proč theProduct? proč ne jen product?

Kopíruji typ i název parametru volané funkce.